dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
While I'm not sure if we will ever have a good full API abstraction for materials and textures, one thing to keep in mind is that a lot of APIs (especially Assimp) are often designed to cater to older rendering systems where lots of things like shadows and transparency were special-cased and needed all sorts of special datatypes.  Much of this gets simplified in newer pipeline-oriented rendering, and if we specifically keep that in mind, we might not need nearly as much cruft.

@mkoval, @mklingen: perhaps we should get an opinion from @PyryM here?